### ISSUE TITLE
Unable to provision vSphere RKE2 cluster, machine provision error "Invalid command line. Found extra arguments"

### ISSUE THREAD

**Issue by bvankampen** [rancher/rancher#33870]
**Rancher Server Setup**
- Rancher version: 2.6 
- Installation option (Docker install/Helm Chart): Helm Chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE2
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: v1.21.3
- Cluster Type (Local/Downstream): Downstream, Infrastructure Provider vSphere 

**Describe the bug**
When deploying a RKE2 cluster on vSphere I get an error in the rancher/machine jobs: 

`Invalid command line. Found extra arguments [--vmwarevsphere-cloud-config=/path/to/machine/files/cloud-config --vmwarevsphere-cpu-count=2 --vmwarevsphere-creation-type=template --vmwarevsphere-datacenter=/O-DC --vmwarevsphere-datastore=/O-DC/datastore/V7000-06-DB-DEV-O-SSD-01 --vmwarevsphere-disk-size=20000 --vmwarevsphere-folder=/O-DC/vm/Kubernetes --vmwarevsphere-hostsystem=__SENTINEL__ --vmwarevsphere-memory-size=2048 --vmwarevsphere-pool=/O-DC/host/Cluster-1/Resources/Rancher-latest --vmwarevsphere-ssh-password=tcuser --vmwarevsphere-ssh-port=22 --vmwarevsphere-ssh-user-group=staff --vmwarevsphere-ssh-user=docker --vmwarevsphere-vcenter-port=443 cluster-test-cluster-test-fzhfn]`

**To Reproduce**
- Create a RKE2 cluster on vSphere. 

```yaml
apiVersion: provisioning.cattle.io/v1
kind: Cluster
metadata:
  annotations:
    field.cattle.io/creatorId: user-c9zv9
  creationTimestamp: "2021-08-02T11:47:33Z"
  finalizers:
  - wrangler.cattle.io/provisioning-cluster-remove
  generation: 2
  managedFields:
  - apiVersion: provisioning.cattle.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:finalizers:
          .: {}
          v:"wrangler.cattle.io/provisioning-cluster-remove": {}
      f:spec:
        .: {}
        f:cloudCredentialSecretName: {}
        f:defaultPodSecurityPolicyTemplateName: {}
        f:kubernetesVersion: {}
        f:rkeConfig:
          .: {}
          f:chartValues:
            .: {}
            f:rke2-calico: {}
          f:etcd:
            .: {}
            f:snapshotRetention: {}
            f:snapshotScheduleCron: {}
          f:localClusterAuthEndpoint: {}
          f:machineGlobalConfig:
            .: {}
            f:cni: {}
            f:profile: {}
          f:machinePools: {}
          f:machineSelectorConfig: {}
          f:upgradeStrategy:
            .: {}
            f:controlPlaneConcurrency: {}
            f:controlPlaneDrainOptions:
              .: {}
              f:timeout: {}
            f:workerConcurrency: {}
            f:workerDrainOptions:
              .: {}
              f:timeout: {}
      f:status:
        .: {}
        f:clusterName: {}
        f:conditions: {}
        f:observedGeneration: {}
    manager: rancher
    operation: Update
    time: "2021-08-02T11:47:33Z"
  name: cluster-test
  namespace: fleet-default
  resourceVersion: "1641964"
  uid: 16a66d14-4a98-4c52-99fa-92b94b676375
spec:
  cloudCredentialSecretName: o-vsca01
  defaultPodSecurityPolicyTemplateName: unrestricted
  kubernetesVersion: v1.21.3-rc1+rke2r2
  rkeConfig:
    chartValues:
      rke2-calico:
        calicoctl:
          image: rancher/mirrored-calico-ctl
          tag: v3.19.1
        certs:
          node:
            cert: null
            commonName: null
            key: null
          typha:
            caBundle: null
            cert: null
            commonName: null
            key: null
        global:
          systemDefaultRegistry: ""
        imagePullSecrets: {}
        installation:
          calicoNetwork:
            bgp: Disabled
            ipPools:
            - cidr: 10.42.0.0/16
              encapsulation: VXLAN
              natOutgoing: Enabled
          controlPlaneTolerations:
          - effect: NoSchedule
            key: node-role.kubernetes.io/control-plane
            operator: Exists
          - effect: NoExecute
            key: node-role.kubernetes.io/etcd
            operator: Exists
          enabled: true
          imagePath: rancher
          imagePrefix: mirrored-calico-
          kubernetesProvider: ""
        tigeraOperator:
          image: rancher/mirrored-calico-operator
          registry: docker.io
          version: v1.17.4
    etcd:
      snapshotRetention: 5
      snapshotScheduleCron: 0 */5 * * *
    localClusterAuthEndpoint: {}
    machineGlobalConfig:
      cni: calico
      profile: null
    machinePools:
    - controlPlaneRole: true
      etcdRole: true
      machineConfigRef:
        kind: VmwarevsphereConfig
        name: nc-cluster-test-cluster-test-2dk47
      name: cluster-test
      quantity: 1
      workerRole: true
    machineSelectorConfig:
    - config:
        cloud-provider-name: none
    upgradeStrategy:
      controlPlaneConcurrency: 10%
      controlPlaneDrainOptions:
        timeout: 0
      workerConcurrency: 10%
      workerDrainOptions:
        timeout: 0
status:
  clusterName: c-m-2df5f888
  conditions:
  - lastUpdateTime: "2021-08-02T11:47:33Z"
    reason: Reconciling
    status: "True"
    type: Reconciling
  - lastUpdateTime: "2021-08-02T11:47:33Z"
    status: "False"
    type: Stalled
  - lastUpdateTime: "2021-08-02T11:47:41Z"
    status: "True"
    type: Created
  - lastUpdateTime: "2021-08-02T11:47:38Z"
    status: "True"
    type: RKECluster
  - lastUpdateTime: "2021-08-02T11:47:33Z"
    status: "True"
    type: BackingNamespaceCreated
  - lastUpdateTime: "2021-08-02T11:47:33Z"
    status: "True"
    type: DefaultProjectCreated
  - lastUpdateTime: "2021-08-02T11:47:33Z"
    status: "True"
    type: SystemProjectCreated
  - lastUpdateTime: "2021-08-02T11:47:38Z"
    message: 'provisioning bootstrap node(s) cluster-test-cluster-test-54668c5594-6r9x2:
      creating server (VmwarevsphereMachine) in infrastructure provider'
    reason: Waiting
    status: Unknown
    type: Provisioned
  - lastUpdateTime: "2021-08-02T11:47:34Z"
    status: "True"
    type: InitialRolesPopulated
  - lastUpdateTime: "2021-08-02T11:47:34Z"
    status: "True"
    type: CreatorMadeOwner
  - status: Unknown
    type: Pending
  - lastUpdateTime: "2021-08-02T11:47:35Z"
    message: Waiting for API to be available
    status: Unknown
    type: Waiting
  - lastUpdateTime: "2021-08-02T11:47:35Z"
    status: "True"
    type: NoDiskPressure
  - lastUpdateTime: "2021-08-02T11:47:35Z"
    status: "True"
    type: NoMemoryPressure
  - lastUpdateTime: "2021-08-02T11:47:41Z"
    status: "False"
    type: Connected
  observedGeneration: 2
```

**Result**
Cluster won't provision. 

**Expected Result**
New RKE2 Downstram Cluster

**Comment by bvankampen** [comment]
The yaml of the machine providor pod:

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    cni.projectcalico.org/podIP: 10.42.2.17/32
    cni.projectcalico.org/podIPs: 10.42.2.17/32
    kubernetes.io/psp: global-unrestricted-psp
  creationTimestamp: "2021-08-02T11:47:37Z"
  generateName: cluster-test-cluster-test-fzhfn-machine-provision-
  labels:
    controller-uid: 6423e310-0a87-4c77-b5df-45901b8d4e76
    job-name: cluster-test-cluster-test-fzhfn-machine-provision
    rke.cattle.io/infra-machine-group: rke-machine.cattle.io
    rke.cattle.io/infra-machine-kind: VmwarevsphereMachine
    rke.cattle.io/infra-machine-name: cluster-test-cluster-test-fzhfn
    rke.cattle.io/infra-machine-version: v1
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:cni.projectcalico.org/podIP: {}
          f:cni.projectcalico.org/podIPs: {}
    manager: calico
    operation: Update
    time: "2021-08-02T11:47:37Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:generateName: {}
        f:labels:
          .: {}
          f:controller-uid: {}
          f:job-name: {}
          f:rke.cattle.io/infra-machine-group: {}
          f:rke.cattle.io/infra-machine-kind: {}
          f:rke.cattle.io/infra-machine-name: {}
          f:rke.cattle.io/infra-machine-version: {}
        f:ownerReferences:
          .: {}
          k:{"uid":"6423e310-0a87-4c77-b5df-45901b8d4e76"}:
            .: {}
            f:apiVersion: {}
            f:blockOwnerDeletion: {}
            f:controller: {}
            f:kind: {}
            f:name: {}
            f:uid: {}
      f:spec:
        f:containers:
          k:{"name":"machine"}:
            .: {}
            f:args: {}
            f:envFrom: {}
            f:image: {}
            f:imagePullPolicy: {}
            f:name: {}
            f:resources: {}
            f:terminationMessagePath: {}
            f:terminationMessagePolicy: {}
            f:volumeMounts:
              .: {}
              k:{"mountPath":"/etc/ssl/certs/ca-additional.pem"}:
                .: {}
                f:mountPath: {}
                f:name: {}
                f:readOnly: {}
                f:subPath: {}
              k:{"mountPath":"/path/to/machine/files"}:
                .: {}
                f:mountPath: {}
                f:name: {}
                f:readOnly: {}
              k:{"mountPath":"/run/secrets/machine"}:
                .: {}
                f:mountPath: {}
                f:name: {}
        f:dnsPolicy: {}
        f:enableServiceLinks: {}
        f:restartPolicy: {}
        f:schedulerName: {}
        f:securityContext: {}
        f:serviceAccount: {}
        f:serviceAccountName: {}
        f:terminationGracePeriodSeconds: {}
        f:volumes:
          .: {}
          k:{"name":"bootstrap"}:
            .: {}
            f:name: {}
            f:secret:
              .: {}
              f:defaultMode: {}
              f:optional: {}
              f:secretName: {}
          k:{"name":"machine-files"}:
            .: {}
            f:name: {}
            f:secret:
              .: {}
              f:defaultMode: {}
              f:items: {}
              f:secretName: {}
          k:{"name":"tls-ca-additional-volume"}:
            .: {}
            f:name: {}
            f:secret:
              .: {}
              f:defaultMode: {}
              f:optional: {}
              f:secretName: {}
    manager: kube-controller-manager
    operation: Update
    time: "2021-08-02T11:47:37Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:status:
        f:conditions:
          k:{"type":"ContainersReady"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
          k:{"type":"Initialized"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:status: {}
            f:type: {}
          k:{"type":"Ready"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
        f:containerStatuses: {}
        f:hostIP: {}
        f:phase: {}
        f:podIP: {}
        f:podIPs:
          .: {}
          k:{"ip":"10.42.2.17"}:
            .: {}
            f:ip: {}
        f:startTime: {}
    manager: kubelet
    operation: Update
    time: "2021-08-02T11:47:49Z"
  name: cluster-test-cluster-test-fzhfn-machine-provision-tch94
  namespace: fleet-default
  ownerReferences:
  - apiVersion: batch/v1
    blockOwnerDeletion: true
    controller: true
    kind: Job
    name: cluster-test-cluster-test-fzhfn-machine-provision
    uid: 6423e310-0a87-4c77-b5df-45901b8d4e76
  resourceVersion: "1642005"
  uid: 63961db2-3cdd-4750-a03c-74ce7fe9ab45
spec:
  containers:
  - args:
    - --driver-download-url=
    - --driver-hash=
    - --secret-namespace=fleet-default
    - --secret-name=cluster-test-cluster-test-fzhfn-machine-state
    - create
    - --driver=vmwarevsphere
    - --custom-install-script=/run/secrets/machine/value
    - --vmwarevsphere-cfgparam=disk.enableUUID=TRUE
    - --vmwarevsphere-clone-from=/O-DC/vm/Debian Template
    - --vmwarevsphere-cloud-config=/path/to/machine/files/cloud-config
    - --vmwarevsphere-cpu-count=2
    - --vmwarevsphere-creation-type=template
    - --vmwarevsphere-datacenter=/O-DC
    - --vmwarevsphere-datastore=/O-DC/datastore/V7000-06-DB-DEV-O-SSD-01
    - --vmwarevsphere-disk-size=20000
    - --vmwarevsphere-folder=/O-DC/vm/Kubernetes
    - --vmwarevsphere-hostsystem=__SENTINEL__
    - --vmwarevsphere-memory-size=2048
    - --vmwarevsphere-pool=/O-DC/host/Cluster-1/Resources/Rancher-latest
    - --vmwarevsphere-ssh-password=tcuser
    - --vmwarevsphere-ssh-port=22
    - --vmwarevsphere-ssh-user-group=staff
    - --vmwarevsphere-ssh-user=docker
    - --vmwarevsphere-vcenter-port=443
    - cluster-test-cluster-test-fzhfn
    envFrom:
    - secretRef:
        name: cluster-test-cluster-test-fzhfn-machine-driver-secret
    image: rancher/machine:v0.15.0-rancher64
    imagePullPolicy: Always
    name: machine
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /path/to/machine/files
      name: machine-files
      readOnly: true
    - mountPath: /run/secrets/machine
      name: bootstrap
    - mountPath: /etc/ssl/certs/ca-additional.pem
      name: tls-ca-additional-volume
      readOnly: true
      subPath: ca-additional.pem
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-dwq9d
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: rancher-mgmt-rke2-latest-3
  preemptionPolicy: PreemptLowerPriority
  priority: 0
  restartPolicy: Never
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: cluster-test-cluster-test-fzhfn-machine-provision
  serviceAccountName: cluster-test-cluster-test-fzhfn-machine-provision
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - name: machine-files
    secret:
      defaultMode: 384
      items:
      - key: cloud-config
        path: cloud-config
      secretName: cluster-test-cluster-test-fzhfn-machine-provision
  - name: bootstrap
    secret:
      defaultMode: 448
      optional: false
      secretName: cluster-test-bootstrap-template-m6r6k-machine-bootstrap
  - name: tls-ca-additional-volume
    secret:
      defaultMode: 256
      optional: true
      secretName: tls-ca-additional-volume
  - name: kube-api-access-dwq9d
    projected:
      defaultMode: 420
      sources:
      - serviceAccountToken:
          expirationSeconds: 3607
          path: token
      - configMap:
          items:
          - key: ca.crt
            path: ca.crt
          name: kube-root-ca.crt
      - downwardAPI:
          items:
          - fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
            path: namespace
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2021-08-02T11:47:37Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2021-08-02T11:47:49Z"
    message: 'containers with unready status: [machine]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2021-08-02T11:47:49Z"
    message: 'containers with unready status: [machine]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2021-08-02T11:47:37Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: containerd://7bcf167eadbb3fa9a77261ab05dbdb52a2811de2f9f093ccdc4105e46c21e36f
    image: docker.io/rancher/machine:v0.15.0-rancher64
    imageID: docker.io/rancher/machine@sha256:c40fb667f3890deba468f8e1ef7d554f337262e673e5d87cc639b59b2eb1c1fe
    lastState: {}
    name: machine
    ready: false
    restartCount: 0
    started: false
    state:
      terminated:
        containerID: containerd://7bcf167eadbb3fa9a77261ab05dbdb52a2811de2f9f093ccdc4105e46c21e36f
        exitCode: 1
        finishedAt: "2021-08-02T11:47:49Z"
        message: |
          Invalid command line. Found extra arguments [--vmwarevsphere-cloud-config=/path/to/machine/files/cloud-config --vmwarevsphere-cpu-count=2 --vmwarevsphere-creation-type=template --vmwarevsphere-datacenter=/O-DC --vmwarevsphere-datastore=/O-DC/datastore/V7000-06-DB-DEV-O-SSD-01 --vmwarevsphere-disk-size=20000 --vmwarevsphere-folder=/O-DC/vm/Kubernetes --vmwarevsphere-hostsystem=__SENTINEL__ --vmwarevsphere-memory-size=2048 --vmwarevsphere-pool=/O-DC/host/Cluster-1/Resources/Rancher-latest --vmwarevsphere-ssh-password=tcuser --vmwarevsphere-ssh-port=22 --vmwarevsphere-ssh-user-group=staff --vmwarevsphere-ssh-user=docker --vmwarevsphere-vcenter-port=443 cluster-test-cluster-test-fzhfn]
        reason: Error
        startedAt: "2021-08-02T11:47:39Z"
  hostIP: 10.41.9.33
  phase: Failed
  podIP: 10.42.2.17
  podIPs:
  - ip: 10.42.2.17
  qosClass: BestEffort
  startTime: "2021-08-02T11:47:37Z"
```

**Comment by kinarashah** [comment]
Thank you for trying out 2.6! 
This issue was fixed recently for https://github.com/rancher/rancher/issues/33867. Should be available in latest `rancher/rancher:v2.6-head` according to https://github.com/rancher/rancher/issues/33867#issuecomment-894425851.

**Comment by bvankampen** [comment]
Fix also works for me, clusters are now provisioning fine.
